### ISSUE TITLE
New version of NetAnalyzers throws CA1420 for delegates

### ISSUE THREAD

**Issue by mattleibow** [dotnet/roslyn-analyzers#6094]
### Analyzer

**Diagnostic ID**: CA1420 `Delegates with managed types as parameters or the return type require runtime marshalling to be enabled in the assembly where the delegate is defined`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.0-preview1.22302.1

### Describe the bug

D:\a\_work\1\s\src\Controls\src\Core\Platform\Tizen\Extensions\DragDropExtensions.cs(89,28): error CA1420: Delegates with managed types as parameters or the return type require runtime marshalling to be enabled in the assembly where the delegate is defined. [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj]
  
I started getting this issue after updating the package from:

 - GOOD: 7.0.0-preview1.22274.2
 - BAD: 7.0.0-preview1.22302.1
 - BAD: 7.0.0-preview1.22379.1

### Steps To Reproduce

1. Clone: https://github.com/dotnet/maui/tree/release/7.0.1xx-preview7
2. Update https://github.com/dotnet/maui/blob/release/7.0.1xx-preview7/eng/Versions.props#L39 to be the newer: 7.0.0-preview1.22302.1
3. `dotnet build Microsoft.Maui.BuildTasks.slnf /p:configuration=Release`

### Expected behavior
Builds.

### Actual behavior
Error.

## Additional context

You may need to install the tizen workload, but I also saw this on other projects. I know this is a vague issue with a "download my code and build" but I am not sure what this error even means. Is it saying that delegates are not good?

I _think_ it seems to be if some p/invoke has a delegate that uses a IntPtr

**Comment by Youssef1313** [comment]
https://github.com/dotnet/runtime/issues/63714 has some info around the analyzer. To my understanding, the analyzer should fire only if you have `System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute` added to the assembly. Does dotnet/maui add this attribute somewhere?

**Comment by mattleibow** [comment]
I do not believe so.

**Comment by amin1best** [comment]
Hi,
Another reproduction:
```powershell
dotnet new console
```
Replace the content of "Program.cs" with:
```cs
using System.Runtime.InteropServices;

class Program
{
    [StructLayout(LayoutKind.Sequential, Pack = 1, CharSet = CharSet.Auto)]
    public struct testStruct
    {
        public CULong testPrp;
    }

    [UnmanagedFunctionPointerAttribute(CallingConvention.StdCall, CharSet = CharSet.Auto)]
    internal delegate CULong fncDlg(ref testStruct strcPrm);
    static void Main()
    {
        var lb = NativeLibrary.Load(@"test.dll");
        var fnc = NativeLibrary.GetExport(lb, "testFnc");

        testStruct strcObj = new testStruct();

        //#pragma warning disable CA1420
        var dlg = Marshal.GetDelegateForFunctionPointer<fncDlg>(fnc);
        //#pragma warning restore CA1420

        dlg(ref strcObj);

        NativeLibrary.Free(lb);
    }
}
```
and
```powershell
dotnet build
```
show:
```
Program.cs(21,19): warning CA1420: Delegates with managed types as parameters or the return type require runtime marshalling to be enabled in the assembly where the delegat 
e is defined.
```
dotnet version: 7.0.100-preview.7.22377.5

**Comment by mattleibow** [comment]
Does it compile with the older nuger package?

**Comment by amin1best** [comment]
In 6.0.100-rc.2.21505.57, it works fine and doesn't warn.
